### PR TITLE
Add LR XML doc

### PR DESCRIPTION
**Pull request by wschin**
Toward #2522 following #3218.

### CODE REVIEW

File: src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LogisticRegression.cs [hunk 1]
```diff
@@ -27,8 +27,55 @@
 namespace Microsoft.ML.Trainers
 {
 
-    /// <include file='doc.xml' path='doc/members/member[@name="LBFGS"]/*' />
```
Comment (shmoradims) [inline src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LogisticRegression.cs]: >file='doc.xml' path='doc/members/member[@name="LBFGS"]/* [](start = 17, length = 56)

please delete these sections that you moved here from doc.xml

File: src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LogisticRegression.cs [hunk 2]
```diff
@@ -27,8 +27,55 @@
...
+    /// The <see cref="IEstimator{TTransformer}"/> to predict a target using a linear logistic regression model trained with L-BFGS method.
+    /// </summary>
+    /// <remarks>
+    /// <format type="text/markdown"><![CDATA[
+    /// To create this trainer, use [LbfgsLogisticRegression](xref:Microsoft.ML.StandardTrainersCatalog.LbfgsLogisticRegression(BinaryClassificationCatalog.BinaryClassificationTrainers, string, string, string, float, float, float, int, bool))
+    /// or [LbfgsLogisticRegression(Options)](xref:Microsoft.ML.StandardTrainersCatalog.LbfgsLogisticRegression(BinaryClassificationCatalog.BinaryClassificationTrainers, LbfgsLogisticRegressionBinaryTrainer.Options)).
+    ///
+    /// [!include[io](~/../docs/samples/docs/api-reference/io-columns-binary-classification.md)]
+    ///
+    /// ### Trainer Characteristics
```
Comment (singlis) [inline src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LogisticRegression.cs]: Should this be Estimator Characteristics?

File: src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LogisticRegression.cs [hunk 3]
```diff
@@ -27,8 +27,55 @@
...
 {
 
-    /// <include file='doc.xml' path='doc/members/member[@name="LBFGS"]/*' />
-    /// <include file='doc.xml' path='docs/members/example[@name="LogisticRegressionBinaryClassifier"]/*' />
+    /// <summary>
+    /// The <see cref="IEstimator{TTransformer}"/> to predict a target using a linear logistic regression model trained with L-BFGS method.
+    /// </summary>
+    /// <remarks>
+    /// <format type="text/markdown"><![CDATA[
+    /// To create this trainer, use [LbfgsLogisticRegression](xref:Microsoft.ML.StandardTrainersCatalog.LbfgsLogisticRegression(BinaryClassificationCatalog.BinaryClassificationTrainers, string, string, string, float, float, float, int, bool))
```
Comment (shmoradims) [inline src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LogisticRegression.cs]: >LbfgsLogisticRegression(BinaryClassificationCatalog.BinaryClassificationTrainers, string, string, string, float, float, float, int, bool)) [](start = 104, length = 138)

this UID won't work, you need fully qualified names. please use [https://xref.docs.microsoft.com/autocomplete?text=LbfgsLogisticRegression](https://xref.docs.microsoft.com/autocomplete?text=LbfgsLogisticRegression)  to get the UID. It's this one:

Microsoft.ML.StandardTrainersCatalog.LbfgsLogisticRegression(Microsoft.ML.BinaryClassificationCatalog.BinaryClassificationTrainers,System.String,System.String,System.String,System.Single,System.Single,System.Single,System.Int32,System.Boolean)

please also fix for the other one.

File: src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LogisticRegression.cs [hunk 4]
```diff
@@ -39,6 +86,10 @@ public sealed partial class LbfgsLogisticRegressionBinaryTrainer : LbfgsTrainerB
         internal const string Summary = "Logistic Regression is a method in statistics used to predict the probability of occurrence of an event and can "
             + "be used as a classification algorithm. The algorithm predicts the probability of occurrence of an event by fitting data to a logistical function.";
 
+        /// <summary>
+        /// Options for the <see cref="LbfgsLogisticRegressionBinaryTrainer"/> as used in
+        /// [LbfgsLogisticRegression(Options)](xref:Microsoft.ML.StandardTrainersCatalog.LbfgsLogisticRegression(BinaryClassificationCatalog.BinaryClassificationTrainers, LbfgsLogisticRegressionBinaryTrainer.Options)).
```
Comment (shmoradims) [inline src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LogisticRegression.cs]: >(xref:Microsoft.ML.StandardTrainersCat [](start = 46, length = 38)

ditto: fix uid

File: src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LogisticRegression.cs (hunk 3 above)
Comment (wschin) [reply src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LogisticRegression.cs]: Thanks. Sure.

---
In reply to: [276468758](https://github.com/dotnet/machinelearning/pull/3385#discussion_r276468758) [](ancestors = 276468758)

File: src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LogisticRegression.cs (hunk 2 above)
Comment (wschin) [reply src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LogisticRegression.cs]: No. I don't see others are doing `Estimator`.

---
In reply to: [276468119](https://github.com/dotnet/machinelearning/pull/3385#discussion_r276468119) [](ancestors = 276468119)

File: src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs [hunk 5]
```diff
@@ -518,11 +518,11 @@ public IClassificationLoss CreateComponent(IHostEnvironment env)
         }
 
         /// <summary>
-        ///  Predict a target using a linear binary classification model trained with the <see cref="Trainers.LbfgsLogisticRegressionBinaryTrainer"/> trainer.
+        ///  Create an <see cref="Trainers.LbfgsLogisticRegressionBinaryTrainer"/>, which predicts a target using a linear binary classification model trained over boolean label data.
```
Comment (codemzs) [inline src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs]: >an [](start = 20, length = 2)

Can you please check with Shahab/@natk if we need "an"? In my PRs it was just Create <see cref..>

Comment (shmoradims) [reply src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs]: "Create **a**" or just "Create"

File: src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LogisticRegression.cs [hunk 6]
```diff
@@ -27,8 +27,55 @@
...
+    ///
+    /// Regularization is a method that can render an ill-posed problem more tractable by imposing constraints that provide information to supplement the data and that prevents overfitting by penalizing model's magnitude usually measured by some norm functions.
+    /// This can improve the generalization of the model learned by selecting the optimal complexity in the bias-variance tradeoff.
+    /// Regularization works by adding the penalty that is associated with coefficient values to the error of the hypothesis.
+    /// An accurate model with extreme coefficient values would be penalized more, but a less accurate model with more conservative values would be penalized less.
+    ///
+    /// This learner supports [elastic net regularization](https://en.wikipedia.org/wiki/Elastic_net_regularization): a linear combination of L1-norm (LASSO), $|| \boldsymbol{w} ||_1$, and L2-norm (ridge), $|| \boldsymbol{w} ||_2^2$ regularizations.
+    /// L1-nrom and L2-norm regularizations have different effects and uses that are complementary in certain respects.
```
Comment (singlis) [inline src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LogisticRegression.cs]: >r [](start = 12, length = 1)

norm

File: src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LogisticRegression.cs [hunk 7]
```diff
@@ -27,8 +27,55 @@
...
+    /// Using L1-norm can increase sparsity of the trained $\boldsymbol{w}$.
+    /// When working with high-dimensional data, it shrinks small weights of irrevalent features to 0 and therefore no reource will be spent on those bad features when making prediction.
```
Comment (singlis) [inline src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LogisticRegression.cs]: >irrevalent [](start = 77, length = 10)

irrelavent

**Pull request merged**